modules/ecash-lib/src/unsignedTx.ts

```diff
@@ -119,6 +219,29 @@ export class UnsignedTxInput {
                 ? redeemScript
                 : redeemScript.cutOutCodesep(nCodesep);
 
+        // Sign LEGACY signatures that don't use SIGHASH_FORKID
+        if (sigHashType.variant == SigHashTypeVariant.LEGACY) {
```

Comment:
any reason for not `sigHashType.variant === SigHashTypeVariant.LEGACY)` ?

`===` vs `==`

Reply:
Good point.